File: FinancialBI/Views/Version Selection/VersionSelectionPane.vb
' CVersionSelectionPane.vb
'
' VIEW for versions selection
'
'
' To do:
'       -
'       -
'
'
' Known bugs:   
'       -
' 
'
' Author: Julien Monnereau
' Last modified: 16/12/2015


Imports System.Runtime.InteropServices
Imports AddinExpress.XL
Imports System.Windows.Forms


Public Class VersionSelectionPane

#Region "Instance Variables"

    Private m_versionSelectionController As VersionSelection
    Private mode As Int32

#End Region

#Region "Initialize"

    Public Sub New()
        MyBase.New()
        InitializeComponent()
        InitializeMultiLanguage()
    End Sub

    Private Sub InitializeMultiLanguage()

        m_versionSelectionLabel.Text = Local.GetValue("general.select_version")
        m_validateButton.Text = Local.GetValue("general.validate")

    End Sub

    Friend Function Init() As Boolean

        m_versionSelectionController = New VersionSelection(m_versionsTreeviewImageList, Me)
        InsertDataVersionSelection()
        AddHandler m_validateButton.Click, AddressOf m_versionSelectionController.SetSelectedVersion
        Return False

    End Function

    Private Sub InsertDataVersionSelection()

        TableLayoutPanel1.Controls.Clear()
        TableLayoutPanel1.Controls.Add(m_versionSelectionController.versionsTV, 0, 1)
        TableLayoutPanel1.GetControlFromPosition(0, 1).Dock = DockStyle.Fill

    End Sub

#End Region

#Region "Interface"

    Public Sub ClearAndClose()

        If Not m_versionSelectionController Is Nothing Then
            m_versionSelectionController.versionsTV.Nodes.Clear()
            TableLayoutPanel1.Controls.Remove(TableLayoutPanel1.GetControlFromPosition(0, 1))
            m_versionSelectionController = Nothing
            Me.Hide()
        End If

    End Sub

    Friend Sub SetVersion(ByRef version_id As String)

        If m_versionSelectionController Is Nothing Then m_versionSelectionController = New VersionSelection(m_versionsTreeviewImageList, Me)
        m_versionSelectionController.versionsTV.SelectedNode = m_versionSelectionController.versionsTV.Nodes.Find(version_id, True)(0)
        m_versionSelectionController.SetSelectedVersion()

    End Sub

#End Region

#Region "Events"

    Private Sub ADXExcelTaskPane1_ADXBeforeTaskPaneShow(sender As Object, e As ADXBeforeTaskPaneShowEventArgs) Handles MyBase.ADXBeforeTaskPaneShow
        Me.Visible = GlobalVariables.VersionsSelectionTaskPaneVisible
    End Sub

    Private Sub CVersionSelectionPane_Deactivate(sender As Object, e As EventArgs) Handles MyBase.Deactivate
        ClearAndClose()
    End Sub


#End Region


End Class

File: Excel FinancialBI/Excel FinancialBI/MVC/View/Version Selection/VersionSelectionPane.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace FBI.MVC.View
{
  using Controller;
  using Utils;
  using Forms;
  using Model.CRUD;
  using Model;
  using Network;

  public partial class VersionSelectionPane : AddinExpress.XL.ADXExcelTaskPane
  {
    public bool m_shown { set; get; }
    public FbiTreeView<Version> m_versionTV;

    public VersionSelectionPane()
    {
      InitializeComponent();
      this.MultilangueSetup();
      m_shown = false;
      m_versionTV = new FbiTreeView<Version>();
      SuscribeEvents();
    }

    void SuscribeEvents()
    {
      m_validateButton.Click += OnValidateButtonClick;
    }

    void OnValidateButtonClick(object p_sender, EventArgs p_e)
    {
      if (m_versionTV.SelectedNode != null)
      {
        Version l_version = VersionModel.Instance.GetValue((UInt32)m_versionTV.SelectedNode.Value);

          if (l_version == null)
            return;
        if (l_version.IsFolder)
        {
          MessageBox.Show(Local.GetValue("versions.error.cannot_select_folder"));
          return;
        }
        Addin.VersionId = l_version.Id;
        Hide();
      }
    }

    void MultilangueSetup()
    {
      m_versionSelectionLabel.Text = Local.GetValue("general.select_version");
      m_validateButton.Text = Local.GetValue("general.validate");
    }

    void VersionSelectionPane_ADXBeforeTaskPaneShow(object sender, AddinExpress.XL.ADXBeforeTaskPaneShowEventArgs e)
    {
      if (m_shown == false)
      {
        this.Visible = false;
      }
      else
      {
        RefreshVersionsTV();
        this.Refresh();
      }
    }

    private void RefreshVersionsTV()
    {
      TableLayoutPanel1.Controls.Remove(m_versionTV);
      m_versionTV = new FbiTreeView<Version>(VersionModel.Instance.GetDictionary(), null, true);
      m_versionTV.ImageList = m_versionsTreeviewImageList;
      TableLayoutPanel1.Controls.Add(m_versionTV, 0, 1);
      m_versionTV.Dock = DockStyle.Fill;
      //m_versionTV.SelectedNode = m_versionTV.FindNode(Properties.Settings.Default.version_id);
    }

  }
}
